packages/maskbook/src/plugins/Trader/UI/trader/TraderDialog.tsx

```diff
@@ -0,0 +1,120 @@
+import { useCallback } from 'react'
+import { CardActions, createStyles, DialogContent, Link, makeStyles, Typography } from '@material-ui/core'
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown'
+import { findIndex } from 'lodash-es'
+import { InjectedDialog } from '../../../../components/shared/InjectedDialog'
+import { useRemoteControlledDialog } from '../../../../utils/hooks/useRemoteControlledDialog'
+import { resolveTradeProviderName } from '../../pipes'
+import { FootnoteMenu, FootnoteMenuOption } from '../trader/FootnoteMenu'
+import { TradeContext, useTradeContext } from '../../trader/useTradeContext'
+import { useCurrentTradeProvider } from '../../trending/useCurrentTradeProvider'
+import { MaskbookTextIcon } from '../../../../resources/MaskbookIcon'
+import { TradeProviderIcon } from '../trader/TradeProviderIcon'
+import { currentTradeProviderSettings } from '../../settings'
+import { getEnumAsArray } from '../../../../utils/enum'
+import { TradeProvider } from '../../types'
+import { PluginTraderMessages } from '../../messages'
+import { Trader } from './Trader'
+
+const useStyles = makeStyles((theme) => {
+    return createStyles({
+        footer: {
+            justifyContent: 'space-between',
+        },
+        footnote: {
+            color: theme.palette.text.secondary,
+            fontSize: 10,
+            marginRight: theme.spacing(0.5),
+        },
+        footLink: {
+            cursor: 'pointer',
+            marginRight: theme.spacing(0.5),
+            '&:last-child': {
+                marginRight: 0,
+            },
+        },
+        footMenu: {
+            color: theme.palette.text.secondary,
+            fontSize: 10,
+            display: 'flex',
+            alignItems: 'center',
+        },
+        footName: {
+            marginLeft: theme.spacing(0.5),
+        },
+        maskbook: {
+            width: 40,
+            height: 10,
+        },
+    })
+})
+
+const tradeProviders = [
```

Comment:
Only one truth rule. Please use `useAvailableTraderProviders` instead.

Reply:
Fixed in https://github.com/DimensionDev/Maskbook/pull/2853/commits/20e26373916df034e63efea39836e131829f4dc1